### PR TITLE
Field error component

### LINKED ISSUE #522: Field error component
Currently form field error cannot be accessed via DOM so it cannot be via cypress
It should be possible to get field error without mouseover

### PR DESCRIPTION
**Pull request by LamaEats**
A brand new component that combine all props of Input and FormInput components

## PR includes

- [x] Feature
- [x] DX

## Related issues

Resolve #522 

## QA Instructions, Screenshots, Recordings
 
image
image
image
image

### CODE REVIEW

File: src/components/FormControl/FormControl.tsx [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+    iconRight?: React.ReactNode;
+}
+
+interface FormControlErrorProps extends Omit<React.ComponentProps<typeof ErrorPopup>, 'err'> {
+    error?: React.ComponentProps<typeof ErrorPopup>['err'];
+}
+
+const FormControlContext = createContext<FormControlContext | void>(undefined);
+
+const fontSizeMap: { [key in Size]: number } = {
```
Comment (awinogradov) [inline src/components/FormControl/FormControl.tsx:77]: as const

File: src/components/FormControl/FormControl.tsx [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    s: 11,
+    m: 14,
+    l: 24,
+};
+
+const gapSizeMap: { [key in Size]: { gap: string; padding: [string, string] } } = {
```
Comment (awinogradov) [inline src/components/FormControl/FormControl.tsx:83]: as const

**Review by awinogradov — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**